python/rmm/_lib/memory_resource.pyx

```diff
@@ -588,25 +596,27 @@ cdef class CallbackMemoryResource(DeviceMemoryResource):
     Parameters
     ----------
     allocate_func: callable
-        The allocation function must accept a single integer argument,
-        representing the number of bytes to allocate, and return an
-        integer representing the pointer to the allocated memory.
+        The allocation function must accept two arguments. An integer
```

Comment:
With the move to cuda::memory_resource there is a third argument: the alignment. I wonder if we should move to make this explicit now. Or just be aware that it may start to be needed.

Reply:
I am wondering whether it makes sense to introduce `rmm::async_device_resource_ref` here

Reply:
@miscco In what sense? As the concept that a memory resource must implement?

I am working towards trying to replace (in the cython wrappers) use of `device_memory_resource *` with `rmm::async_device_resource_ref` but that is taking more time than I had hope (there's a lot of string to unwind).

Reply:
> With the move to cuda::memory_resource there is a third argument: the alignment. I wonder if we should move to make this explicit now. Or just be aware that it may start to be needed.

I can do that here I think.

Reply:
https://github.com/rapidsai/rmm/issues/1499

Reply:
> I can do that here I think. Hmm, the allocate/deallocate pair accept (in addition to bytes) stream xor alignment.

That's probably because you are looking at RMM MRs, which are in transition, not the concepts from cuda::mr. In the latter you can have a stream only in `allocate_async`, never in `allocate`.

Reply:
OK, I decided to leave addressing #1499 as part of migrating properly to `resource_ref`.